exercises/simple-linked-list/src/lib.rs

```diff
@@ -0,0 +1,50 @@
+pub struct SimpleLinkedList<T> {
+    // Delete this field
+    // _dummy is needed to avoid unused parameter error during compilation
+    _dummy: T,
+}
+
+impl<T> SimpleLinkedList<T> {
+    pub fn new() -> Self {
+        unimplemented!()
+    }
+
+    pub fn len(&self) -> usize {
+        unimplemented!()
+    }
+
+    pub fn push(&mut self, element: T) {
+        unimplemented!()
+    }
+
+    pub fn pop(&mut self) -> Option<T> {
+        unimplemented!()
+    }
+
+    pub fn peek(&self) -> Option<&T> {
+        unimplemented!()
+    }
+
+    pub fn rev(&self) -> SimpleLinkedList<T> {
+        unimplemented!()
+    }
+}
+
+
+impl<'a, T> From<&'a [T]> for SimpleLinkedList<T>
+where
+    T: Clone,
+{
+    fn from(item: &[T]) -> Self {
+        unimplemented!()
+    }
+}
+
+impl<T> Into<Vec<T>> for SimpleLinkedList<T>
+where
+    T: Clone,
+{
+    fn into(self) -> Vec<T> {
```

Comment:
If `into` consumes `self`, then we don't need the trait bound `T: Clone`; we can just return the original objects.

Reply:
I agree that T doesn't need to be clonable in this function, but I cannot omit the trait bound here because then I get the error `the trait bound 'T: std::clone::Clone' is not satisfied`.
But you're right: in my example.rs it's not necessary to clone the data.

Reply:
I can't replicate that error using only `src/lib.rs`.

However, I see where it's hit you in `example.rs`. The thing is, you've got `T: Clone` bounds in several places where they're not actually necessary. [Playground example](https://play.rust-lang.org/?gist=2f0b2301b793e9c59185109aae25cd06&version=stable) where they're removed and everything runs successfully.

In order to provide a good example for the students, let's omit the `T: Clone` bound here.

Reply:
Okay, thank you very much for this playgrund example. I didn't know there can be several impl blocks for the same struct with different trait bounds. Now it's clear. I'm changing it and hope everything will be alright then.